src/GoogleUtils.js

```diff
@@ -141,15 +142,43 @@ class GoogleUtils {
       auth: this.oAuth2Client
     })
 
-    const response = await peopleAPI.people.updateContact({
-      resourceName,
-      requestBody: {
-        ...person,
-        etag
-      },
-      updatePersonFields: Object.keys(person).join(',')
-    })
-    return response.data
+    try {
+      const response = await peopleAPI.people.updateContact({
+        resourceName,
+        requestBody: {
+          ...person,
+          etag
+        },
+        updatePersonFields: Object.keys(person).join(',')
+      })
+
+      return response.data
+    } catch (err) {
+      if (
+        err.code === 400 &&
+        err.message.includes(
+          'Request person.etag is different than the current person.etag'
+        )
+      ) {
+        // ask for the new etag and retry
+        const response = await peopleAPI.people.get({
+          resourceName,
+          personFields: 'names'
+        })
+        const newEtag = get(response, 'data.etag')
+        const newResponse = await peopleAPI.people.updateContact({
```

Comment:
Not  abig deal but I guess we could do `return this. updateContact(person, resourceName, newEtag)` to avoid some duplication.

Reply:
That's what I did at first but I had to introduce a `retryEtag` param to avoid potential infinite recursive calls, changed it just before to commit but not sure it's better though...